logical/plugin/plugin.go

```diff
@@ -0,0 +1,75 @@
+package plugin
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/helper/pluginutil"
+	"github.com/hashicorp/vault/logical"
+)
+
+// PluginMap should be used by clients for map of plugins
+var PluginMap = map[string]plugin.Plugin{
+	"backend": &BackendPlugin{},
+}
+
+// NewBackend will return an instance of an RPC-based client implementation of the backend for
+// external plugins, or a concrete implementation of the backend if it is a builtin backend.
+// The backend is returned as a logical.Backend interface.
+func NewBackend(pluginName string, sys pluginutil.LookRunnerUtil) (logical.Backend, error) {
+	// Look for plugin in the plugin catalog
+	pluginRunner, err := sys.LookupPlugin(pluginName)
+	if err != nil {
+		return nil, err
+	}
+
+	var backend logical.Backend
+	if pluginRunner.Builtin {
```

Comment:
Maybe. Not yet, but one can imagine that future backends with minimal performance requirements could be written as plugins so they could easily be enhanced/swapped out by users.